UnityProject/Assets/Scripts/Weapons/Grenade.cs

```diff
@@ -96,6 +96,8 @@ private IEnumerator TimeExplode(GameObject originator)
 		{
 			timerRunning = true;
 			PlayPinSFX(originator.transform.position);
+			UIManager.Action.Throw();
```

Comment:
Only problem is this gets only called Server side so for Clients connecting to the server this is not triggered properly

Reply:
https://github.com/unitystation/unitystation/wiki/Implementing-Interactions#how-do-i-inform-the-client-what-happened 

Should explain it a little bit

Reply:
Have a look at the `ExplodeWhenWelded` example in the IF2 docs. It explains what each function does and you should be able to implement something similar for the grenade: https://github.com/unitystation/unitystation/wiki/Implementing-Interactions#how-do-i-implement-an-interaction